### PR TITLE
refactor(app): disable LPC CTA when empty protocol is uploaded

### LINKED ISSUE #9041: Disable LPC CTA if not at least one labware and one pipette
# Overview 

We need to disable the main Labware Position Check CTA on the Labware Setup Step if the protocol doesn't include at least one labware AND one pipette.

There are a number of maintenance procedures that use the protocol upload without the intent to interact with the pipettes or labware. This should protect users from accidentally white-screening were they to upload one of these protocols and launch LPC.

 In addition to disabling the CTA, we should include a disabled reason that renders on hover of the disabled button.  The tooltip text can read "Labware Position Check requires that the protocol loads labware and pipettes"

### PR DESCRIPTION
**Pull request by sakibh**
# Overview

This PR will disable the LPC CTA on the Labware Setup Step if the protocol doesn't include at least one labware AND one pipette. closes #9041

Screen Shot 2021-12-09 at 3 33 05 PM

# Changelog

- Added boolean to check if `protocolData` contains labware and pipettes, in order to disable the LPC CTA and display a tool tip.
- Added test to check if LPC CTA is disabled with protocolData for pipettes and labware is empty.

# Review requests

- Check that the LPC CTA is disabled with the correct tool tip message when a protocol that doesn't include at least one labware AND one pipette is uploaded.

# Risk assessment

low

### CODE REVIEW

File: app/src/organisms/ProtocolSetup/RunSetupCard/LabwareSetup/index.tsx [hunk 1]
```diff
@@ -207,6 +215,15 @@ export const LabwareSetup = (): JSX.Element | null => {
                   }
                 </Tooltip>
               ) : null}
+              {disableLabwarePositionCheckEmptyProtocols ? (
+                <Tooltip {...tooltipProps}>
+                  {
+                    <Box width={SIZE_5} textAlign={TEXT_ALIGN_CENTER}>
+                      {t('labware_position_check_not_available_empty_protocol')}
```
Comment (b-cooper) [inline app/src/organisms/ProtocolSetup/RunSetupCard/LabwareSetup/index.tsx]: Just curious, is this `Box` strictly required? What happens if you just pass the `{t('labware_position_check_not_available_empty_protocol')}` as the children of the `<Tooltip >`

File: app/src/organisms/ProtocolSetup/RunSetupCard/LabwareSetup/index.tsx [hunk 2]
```diff
@@ -64,6 +66,9 @@ export const LabwareSetup = (): JSX.Element | null => {
   const runStatus = useRunStatus()
   const disableLabwarePositionCheck =
     runStatus != null && runStatus !== RUN_STATUS_IDLE
+  const { protocolData } = useProtocolDetails()
+  const disableLabwarePositionCheckEmptyProtocols =
+    isEmpty(protocolData?.pipettes) && isEmpty(protocolData?.labware)
```
Comment (b-cooper) [inline app/src/organisms/ProtocolSetup/RunSetupCard/LabwareSetup/index.tsx]: I think technically this needs to `isEmpty(protocolData?.pipettes) || isEmpty(protocolData?.labware)` . LPC is meaningless is either of them is empty

File: app/src/organisms/ProtocolSetup/RunSetupCard/LabwareSetup/index.tsx [hunk 3]
```diff
@@ -64,6 +66,9 @@ export const LabwareSetup = (): JSX.Element | null => {
   const runStatus = useRunStatus()
   const disableLabwarePositionCheck =
     runStatus != null && runStatus !== RUN_STATUS_IDLE
+  const { protocolData } = useProtocolDetails()
+  const disableLabwarePositionCheckEmptyProtocols =
```
Comment (b-cooper) [inline app/src/organisms/ProtocolSetup/RunSetupCard/LabwareSetup/index.tsx]: Now that we've got two possible reasons that the LPC CTA could be disabled, a cleaner pattern to use is `disabledReason: string | null` . We use this in `app/src/App/hooks` to establish the reason behind disabling certain navigation items.  Basically, here we'd declare a `let lpcDisabledReason: string | null = null` and then follow with if statement for the two cases 
```
if (runStatus != null && runStatus !== RUN_STATUS_IDLE){
  lpcDisabledReason = t(...)
} else if (isEmpty(protocolData?.pipettes) && isEmpty(protocolData?.labware)) {
  lpcDisabledReason = t(...)
}
```

Then later on we only need to have one tooltip and we render it as long as `lpcDisabledReason !== null`. Same for the button's `disabled` prop.

**Review by b-cooper — CHANGES_REQUESTED** [review]
Looking good. Left a few comments

File: app/src/organisms/ProtocolSetup/RunSetupCard/LabwareSetup/index.tsx (hunk 3 above)
Comment (sakibh) [reply app/src/organisms/ProtocolSetup/RunSetupCard/LabwareSetup/index.tsx]: Thanks for suggesting this. Great way to DRY this up!

File: app/src/organisms/ProtocolSetup/RunSetupCard/LabwareSetup/index.tsx (hunk 2 above)
Comment (sakibh) [reply app/src/organisms/ProtocolSetup/RunSetupCard/LabwareSetup/index.tsx]: Great catch on this! You are right that it definitely should be OR.

File: app/src/organisms/ProtocolSetup/RunSetupCard/LabwareSetup/index.tsx (hunk 1 above)
Comment (sakibh) [reply app/src/organisms/ProtocolSetup/RunSetupCard/LabwareSetup/index.tsx]: Yep, it works just fine without the `Box`.

**Review by b-cooper — APPROVED** [review]
Looks awesome 😎

**Pull request merged**